web/common/useUpdateBusiness.js

```diff
@@ -0,0 +1,43 @@
+import { CHANGE_EMPLOYEE_BUSINESS_TYPE } from "common/utils/apiQueries";
+import { ADMIN_ACTIONS } from "../admin/store/reducers/root";
+import { useAdminStore } from "../admin/store/store";
+import { useSnackbarAlerts } from "./Snackbar";
+import { formatApiError } from "common/utils/errors";
+import { useApi } from "common/utils/api";
+
+export const useUpdateEmployeeBusinessType = (employmentId, companyId) => {
+  const api = useApi();
+  const adminStore = useAdminStore();
+  const alerts = useSnackbarAlerts();
+
+  const udpateEmployeeBusinessType = async businessType => {
+    try {
+      const apiResponse = await api.graphQlMutate(
+        CHANGE_EMPLOYEE_BUSINESS_TYPE,
+        {
+          employmentId,
+          businessType
+        }
+      );
+      const {
+        employments
+      } = apiResponse?.data?.employments?.changeEmployeeBusinessType;
```

Comment:
:orange_circle: Cette ligne génère une erreur dans Sonar (d'où le fait que la CI fail) : 
> Unsafe usage of optional chaining. If it short-circuits with 'undefined' the evaluation will throw TypeError.

```suggestion
      const employments = apiResponse?.data?.employments?.changeEmployeeBusinessType?.employments;
```